### ISSUE TITLE
struct-tags: add support for `cbor` tag

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1476]
https://github.com/fxamacker/cbor

This tag is used in various repository

https://github.com/search?q=language%3Ago%20%2F%60cbor%3A%22%2F&type=code
 
EDIT: Also
 
> A check that forbids `omitempty` and `omitzero` on structs with `cbor:"toarray"` would be useful.
> 
> From https://github.com/fxamacker/cbor/#smaller-encodings-with-struct-tag-options
> 
> > NOTE: When a struct uses toarray, the encoder will ignore omitempty and omitzero to prevent position of encoded array elements from changing. This allows decoder to match encoded elements to their Go struct field.
> 
Source:  
@fxamacker
https://github.com/mgechev/revive/issues/1476#issuecomment-3240571779

**Comment by chavacava** [comment]
Some checks the rule could implement (beyond verifying options are known):

1. when the tag name is `-` then it should be no options. For example, in `cbor:"-,omitempty"`, `omitempty` is useless.
2. the tag for the field `_` (or `_.*`) should have no field name. For example, in 
```go
type xferCBOR struct {
_    struct{} `cbor:"fieldName,toarray"`
```
the rule should spot the tag and propose to write as `cbor:",toarray"`

3. if the tag uses the `keyasint` option, the tag shall have an **integer** as field name.
```go
// Not Ok
Inputs  string  `cbor:",keyasint"`
Outputs string  `cbor:"inputs,keyasint"`
Errors  string `cbor:"errors,optempty,keyasint"`

// Ok
Inputs  string  `cbor:"1,keyasint"`
Outputs string  `cbor:"-100,keyasint"`
Errors  string `cbor:"-1,optempty,keyasint"`
```

4. integers used as fieldnames shall only be allowed if the option `keyasint` is present
```go
// Not Ok
Inputs  string  `cbor:"1,omitempty"`
Outputs string  `cbor:"-12,toarray"`

// Ok
Inputs  string  `cbor:"inputs,omitempty"`
Outputs string  `cbor:",toarray"`
```

5. integers used as fieldnames shall be unique at struct level
```go
// Not Ok
type TxBody struct {
	Inputs  []*TxInput  `cbor:"0,keyasint"`
	Outputs []*TxOutput `cbor:"1,keyasint"`
	Fee     Coin        `cbor:"1,keyasint"` // 1 already used as key in the previous field
}
// Ok
type TxBody struct {
	Inputs  string    `cbor:"0,keyasint"`
	Outputs string   `cbor:"1,keyasint"`
	Fee     Coin        `cbor:"2,keyasint"`
}

// of course, the following is Ok
type TxBody struct {
	Inputs  string  `cbor:"0,keyasint"`
	Outputs string  `cbor:"1,keyasint"`
}

type RxBody struct {
	Signal string  `cbor:"0,keyasint"`
	Error string  `cbor:"1,keyasint"`
}
```

**Comment by fxamacker** [comment]
A check that forbids `omitempty` and `omitzero` on structs with `cbor:"toarray"` would be useful.

From https://github.com/fxamacker/cbor/#smaller-encodings-with-struct-tag-options

> NOTE: When a struct uses toarray, the encoder will ignore omitempty and omitzero to prevent position of encoded array elements from changing. This allows decoder to match encoded elements to their Go struct field.
